src/lib/schedule.c

```diff
@@ -25,459 +25,97 @@
-	if (task->state == TASK_STATE_PENDING) {
-		trace_schedule("_schedule_task(), task already pending");
-		spin_unlock_irq(&sch->lock, flags);
-		return 0;
-	}
-
-	/* is task already queued ? - not enough MIPS to complete ? */
-	if (task->state == TASK_STATE_QUEUED) {
-		trace_schedule("_schedule_task(), task already queued");
-		spin_unlock_irq(&sch->lock, flags);
-		return 0;
-	}
-
-	/* get the current time */
-	current = platform_timer_get(platform_timer);
-
-	ticks_per_ms = clock_ms_to_ticks(sch->clock, 1);
-
-	/* calculate start time - TODO: include MIPS */
-	if (start == 0)
-		task->start = current;
-	else
-		task->start = task->start + ticks_per_ms * start / 1000 -
-			PLATFORM_SCHEDULE_COST;
-
-	/* calculate deadline - TODO: include MIPS */
-	task->deadline = task->start + ticks_per_ms * deadline / 1000;
-
-	/* add task to list */
-	list_item_append(&task->list, &sch->list);
-	task->state = TASK_STATE_QUEUED;
-	spin_unlock_irq(&sch->lock, flags);
-
-	return 1;
-}
-
-/*
- * Add a new task to the scheduler to be run and define a scheduling
- * deadline in time for the task to be ran. Do not invoke the scheduler
- * immediately to run task, but wait intil schedule is next called.
- *
- * deadline is in microseconds relative to start.
- */
-void schedule_task_idle(struct task *task, uint64_t deadline)
-{
-	_schedule_task(task, 0, deadline);
-}
-
-/*
- * Add a new task to the scheduler to be run and define a scheduling
- * window in time for the task to be ran. i.e. task will run between start and
- * deadline times.
- *
- * start is in microseconds relative to last task start time.
- * deadline is in microseconds relative to start.
- */
-void schedule_task(struct task *task, uint64_t start, uint64_t deadline)
-{
-	int need_sched;
-
-	need_sched = _schedule_task(task, start, deadline);
-
-	/* need to run scheduler if task not already running */
-	if (need_sched) {
-		/* rerun scheduler */
-		schedule();
-	}
-}
-
-/* Remove a task from the scheduler when complete */
-void schedule_task_complete(struct task *task)
-{
-	struct schedule_data *sch = *arch_schedule_get();
-	uint32_t flags;
-
-	tracev_schedule("schedule_task_complete()");
-
-	spin_lock_irq(&sch->lock, flags);
-
-	/* Some high priority HW based IRQ handlers can reschedule tasks
-	 * immediately. i.e. before the task context can change task state
-	 * back to COMPLETED. Check here to make sure we dont clobber
-	 * task->state for regular non IRQ users.
-	 */
-	switch (task->state) {
-	case TASK_STATE_RUNNING:
-		task->state = TASK_STATE_COMPLETED;
-		break;
-	case TASK_STATE_QUEUED:
-	case TASK_STATE_PENDING:
-		/* nothing to do here, high priority IRQ has scheduled us */
+	switch (task->type) {
+	case SOF_EDF_TASK:
+		ret = schedule_edf_task_cancel(task);
 		break;
-	default:
-		trace_error(TRACE_CLASS_SCHEDULE, "unexpected task state %d "
-			"at task completion",
-			task->state);
-		task->state = TASK_STATE_COMPLETED;
+	case SOF_LOW_LATENCY_TASK:
+		ret = ll_cancel_default(task);
```

Comment:
Let's unify names with edf: schedule_ll_task_cancel.
Valid for all other functions.

Reply:
done